pkgs/by-name/me/mealie/package.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, nixosTests
+, python3
+, writeShellScript
+}:
+
+let
+  version = "1.1.0";
+  src = fetchFromGitHub {
+    owner = "mealie-recipes";
+    repo = "mealie";
+    rev = "v${version}";
+    sha256 = "sha256-k2jMGJ6bdhkXootWl9cEdevRrGfggmHbkJZHZmhNLTM=";
+  };
+
+  frontend = callPackage (import ./mealie-frontend.nix src version) { };
+
+  python = python3.override {
+    packageOverrides = self: super: {
+      pydantic = self.pydantic_1;
+    };
+  };
+
+  # TODO  Add crfpp in runtime dependencies for ingredient quantities NLP
+
+in python.pkgs.buildPythonPackage rec {
+  pname = "mealie";
+  inherit version src;
+  pyproject = true;
+
+  patches = [
+    # Replace hardcoded paths in code with environment variables (meant for inside Docker only)
+    # So we can configure easily where the data is stored on the server
+    ./mealie_statedir.patch
+    ./mealie_init_db.patch
+    ./mealie_logger.patch
+  ];
+
+  nativeBuildInputs = [
+    python.pkgs.poetry-core
+    python.pkgs.pythonRelaxDepsHook
+  ];
+
+  pythonRelaxDeps = true;
+
+  propagatedBuildInputs = with python.pkgs; [
+    aiofiles
+    alembic
+    aniso8601
+    appdirs
+    apprise
+    bcrypt
+    extruct
+    fastapi
+    gunicorn
+    html2text
+    httpx
+    jinja2
+    lxml
+    orjson
+    passlib
+    pillow
+    psycopg2
+    pyhumps
+    pytesseract
+    python-dotenv
+    python-jose
+    python-ldap
+    python-multipart
+    python-slugify
+    pyyaml
+    rapidfuzz
+    recipe-scrapers
+    sqlalchemy
+    tzdata
+    uvicorn
+  ];
+
+  doCheck = true;
+
+  postInstall = let
+    start_script = writeShellScript "start-mealie" ''
+      export STATIC_FILES="${frontend}"
+      ${lib.getExe python.pkgs.gunicorn} "$@" -k uvicorn.workers.UvicornWorker mealie.app:app;
+    '';
+  in ''
+    mkdir -p $out/config $out/bin
+    substitute ${src}/alembic.ini $out/config/alembic.ini \
+      --replace 'script_location = alembic' 'script_location = ${src}/alembic'
+
+    rm -f $out/bin/*
+    ln -s $out/${python.sitePackages}/mealie/db/init_db.py $out/bin/init_db.py
```

Comment:
The init_db script should probably not be in bin.

I'd put it in libexec, wrap it with the interpreter and expose it via a passtrhu attr.

Reply:
Fair enough, I just didn't find a way to wrap it properly (as it's dependent on some Python imports relative to script path).
Sorry I'll need an example of how to do this

Reply:
Use `makeWrapper` and set the `PYTHONPATH`.